services/port_manager/src/main/java/com/futurewei/alcor/portmanager/proxy/ElasticIpManagerProxy.java

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+
+package com.futurewei.alcor.portmanager.proxy;
+
+import com.futurewei.alcor.common.utils.SpringContextUtil;
+import com.futurewei.alcor.web.entity.elasticip.*;
+import com.futurewei.alcor.web.restclient.ElasticIpManagerRestClient;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ElasticIpManagerProxy {
+
+    private String projectId;
+
+    private ElasticIpManagerRestClient elasticIpManagerRestClient;
+
+    public ElasticIpManagerProxy(String projectId) {
+        this.projectId = projectId;
+        elasticIpManagerRestClient = SpringContextUtil.getBean(ElasticIpManagerRestClient.class);
+    }
+
+    /**
+     * Update the elastic ip association info when port deleted or ip address deleted
+     * @param arg1 Port id
+     * @param arg2 Private ip
+     * @return List<ElasticIpInfoWrapper></>
+     * @throws Exception Rest request exception
+     */
+    public List<ElasticIpInfoWrapper> portIpDeleteEventProcess(Object arg1, Object arg2) throws Exception {

```

Comment:
We could perf testing this method especially when elasticIps = elasticIpManagerRestClient.getElasticIps(this.projectId, filters) returns a lot of eips. 

It involves at least two self-calls which are not parallel at this time.

Reply:
self-calls means extra added calls compared with one integrated controller?

Reply:
Correct. It trigger calls to its own API of EIP mgr, and the call might get load balanced to other instances. This is a well-accepted pattern, just need to pay attention to its performance impact :-)

Reply:
I read your other comments and now I understand your point. This proxy is in Port Mgr so there is no self-call. Never mind.